src/libraries/Microsoft.NETCore.Platforms/src/AssemblyResolver.cs

```diff
@@ -27,6 +28,8 @@ public static void Enable()
             // has run.
         }
 
+        [UnconditionalSuppressMessage("Single file", "IL3000:Avoid accessing Assembly file path when publishing as a single file",
+            Justification = "The code has a fallback to use AppDomain.CurrentDomain.BaseDirectory so is not single file dangerous")]
```

Comment:
```suggestion
            Justification = "The code has a fallback to use AppDomain.CurrentDomain.BaseDirectory so it will work correctly in single-file")]
```